### PR TITLE
 EPMRPP-37762/EPMRPP-37916/EPMRPP-37925/EPMRPP-37926 React. Log page. Attachments rewrite

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx [hunk 1]
```diff
@@ -5,40 +5,96 @@ import { injectIntl, intlShape } from 'react-intl';
...
   };
 
+  componentDidMount() {
+    !this.props.attachments.length && this.props.fetchAttachmentsAction();
+  }
+
   onClickItem = (itemIndex) => this.props.openAttachmentAction(this.props.attachments[itemIndex]);
 
-  onClickThumb = () => {
+  getAttachmentsContent = () => {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx]: Fixed

File: app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/logItemInfoTabs.jsx [hunk 2]
```diff
@@ -107,11 +64,12 @@ export class LogItemInfoTabs extends Component {
 
   state = {
     activeTab: null,
+    activeAttachmentId: null,
   };
 
   componentDidUpdate(prevProps) {
     if (prevProps.logItem.id !== this.props.logItem.id) {
-      this.setActiveTab();
+      this.setInitialState();
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/logItemInfoTabs.jsx]: We need to compare lodItem's `id`.
Seems that in this case we will need to hold previous `logItem.id` in state.. Is it expected?

[the remark this answers is not in the document]
Comment (DiscoElevator) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/logItemInfoTabs.jsx]: Yeas, it's a common case when using `getDerivedStateFromProps`

File: app/src/controllers/log/sagas.js [hunk 3]
```diff
@@ -72,7 +73,11 @@ function* fetchHistoryItemData() {
   const activeLogId = yield select(activeLogIdSelector);
   const prevActiveLogId = yield select(prevActiveLogIdSelector);
   if (activeLogId !== prevActiveLogId) {
-    yield all([call(fetchLogItems), call(fetchActivity)]);
+    yield all([
+      call(fetchLogItems),
+      call(fetchActivity),
+      put(fetchSuccessAction(ATTACHMENTS_NAMESPACE, { content: [] })),
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/log/sagas.js]: Done

File: app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/logItemInfoTabs.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/logItemInfoTabs.jsx]: Ok, fixed.

File: app/src/controllers/log/attachments/sagas.js [hunk 4]
```diff
@@ -28,6 +29,10 @@ function* fetchAttachments() {
   );
 }
 
+function* clearAttachments() {
+  yield put(fetchSuccessAction(ATTACHMENTS_NAMESPACE, { content: [] }));
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/log/attachments/sagas.js]: Fixed

**Pull request merged**